persist/fs/write.go

```diff
@@ -117,35 +120,83 @@ func NewWriter(opts Options) (FileSetWriter, error) {
 // Open initializes the internal state for writing to the given shard,
 // specifically creating the shard directory if it doesn't exist, and
 // opening / truncating files associated with that shard for writing.
-func (w *writer) Open(
-	namespace ident.ID,
-	blockSize time.Duration,
-	shard uint32,
-	blockStart time.Time,
-) error {
-	shardDir := ShardDirPath(w.filePathPrefix, namespace, shard)
-	if err := os.MkdirAll(shardDir, w.newDirectoryMode); err != nil {
-		return err
-	}
-	w.blockSize = blockSize
+func (w *writer) Open(opts WriterOpenOptions) error {
+	var (
+		nextSnapshotIndex int
+		err               error
+		namespace         = opts.Identifier.Namespace
+		shard             = opts.Identifier.Shard
+		blockStart        = opts.Identifier.BlockStart
+	)
+
+	w.blockSize = opts.BlockSize
 	w.start = blockStart
+	w.snapshotTime = opts.Snapshot.SnapshotTime
 	w.currIdx = 0
 	w.currOffset = 0
-	w.checkpointFilePath = filesetPathFromTime(shardDir, blockStart, checkpointFileSuffix)
 	w.err = nil
 
+	var (
+		shardDir            string
+		infoFilepath        string
+		indexFilepath       string
+		summariesFilepath   string
+		bloomFilterFilepath string
+		dataFilepath        string
+		digestFilepath      string
+	)
+	switch opts.FilesetType {
+	case persist.FilesetSnapshotType:
+		shardDir = ShardSnapshotsDirPath(w.filePathPrefix, namespace, shard)
+		// Can't do this outside of the switch statement because we need to make sure
+		// the directory exists before calling NextSnapshotFileIndex
+		if err := os.MkdirAll(shardDir, w.newDirectoryMode); err != nil {
+			return err
+		}
+
+		// This method is not thread-safe, so its the callers responsibilities that they never
```

Comment:
+1 for the explanation. could you put an equivalent note in the `types.go` comment for the method too

Reply:
done